prometheus/mixins.libsonnet

```diff
@@ -52,27 +57,76 @@
     },
   },
 
-  // Legacy Extension points for adding alerts, recording rules and prometheus config.
-  local emptyMixin = {
-    prometheusAlerts+:: {},
-    prometheusRules+:: {},
+  withMixinsConfigmaps(mixins):: {
+    local configMap = k.core.v1.configMap,
+    prometheus_config_maps_mixins+: [
+      local mixin = mixins[mixinName] {
+        prometheusAlerts+:: {},
+        prometheusRules+:: {},
+      };
```

Comment:
Nit: some repetition here, might consider breaking this out into a local "emptyMixin".